src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GCStaticsPreInitDataNode.cs

```diff
@@ -0,0 +1,108 @@
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Contains all GC static fields for a particular EEType.
+    /// Fields that have preinitialized data are pointer reloc pointing to frozen objects.
+    /// Other fields are initialized with 0.
+    /// We simply memcpy these over the GC static EEType object.
+    /// </summary>
+    public class GCStaticsPreInitDataNode : ObjectNode, IExportableSymbolNode
+    {
+        private MetadataType _type;
+        private List<FieldDesc> _sortedPreInitFields;
+
+        public GCStaticsPreInitDataNode(MetadataType type, List<FieldDesc> sortedPreInitFields)
+        {
+            Debug.Assert(!type.IsCanonicalSubtype(CanonicalFormKind.Specific));
+            _type = type;
+            _sortedPreInitFields = sortedPreInitFields;
+        }
+
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(GetMangledName(_type, nameMangler));
+        }
+
+        public int Offset => 0;
+        public MetadataType Type => _type;
+
+        public static string GetMangledName(TypeDesc type, NameMangler nameMangler)
+        {
+            return nameMangler.NodeMangler.GCStatics(type) + "__PreInitData";
+        }
+
+        public virtual bool IsExported(NodeFactory factory) => factory.CompilationModuleGroup.ExportsType(Type);
+
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            // relocs has all the dependencies we need
+            return null;
+        }
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        public override ObjectNodeSection Section => ObjectNodeSection.DataSection;
+        public override bool IsShareable => EETypeNode.IsTypeNodeShareable(_type);
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            // We only need this for CoreRT (at least for now) as we emit static field value directly in GCStaticsNode for N
+            Debug.Assert(factory.Target.Abi == TargetAbi.CoreRT);
+
+            return GetDataForPreInitDataField(
+                this, _type, _sortedPreInitFields, 
+                factory.Target.LayoutPointerSize.AsInt,     // CoreRT static size calculation includes EEType - skip it
+                factory, relocsOnly);
+        }
+
+        public static ObjectData GetDataForPreInitDataField(
+            ISymbolDefinitionNode node, 
+            MetadataType _type, List<FieldDesc> sortedPreInitFields,
+            int startOffset,
+            NodeFactory factory, bool relocsOnly = false)
+        {
+            ObjectDataBuilder builder = new ObjectDataBuilder(factory, relocsOnly);
+
+            builder.RequireInitialAlignment(_type.GCStaticFieldAlignment.AsInt);
```

Comment:
The gc safe memory copying routine is permitted to rely on aligned buffers to determine if it needs to operate in a GC safe manner. Its best to keep this aligned.